libcore/ListModel.vala

```diff
@@ -0,0 +1,323 @@
+        GLib.SequenceIter<GOF.File> seq_iter = files.get_begin_iter ();
+        while (!seq_iter.is_end ()) {
+            GOF.File file_a = seq_iter.@get ();
+
+            @foreach ((model, path, iter) => {
+                GOF.File? file_b = null;
+                model.@get (iter, FM.ColumnID.FILE_COLUMN, out file_b);
+                if (file_a == file_b) {
+                    var row_ref = new Gtk.TreeRowReference (model, path);
+                    rows_to_remove.prepend (row_ref);
+                    seq_iter = seq_iter.next ();
+
+                    if (seq_iter.is_end ()) {
+                        return true;
+                    } else {
+                        file_a = seq_iter.@get ();
+                    }
+                }
+
+                return false;
+            });
+
+            seq_iter = seq_iter.next (); /* file was not in model */
+        }
+
+        bool valid = false;
+        foreach (Gtk.TreeRowReference row_ref in rows_to_remove) {
+            Gtk.TreeIter iter;
+            if (get_iter (out iter, row_ref.get_path ())) {
+                valid = remove (ref iter);
+            }
+        }
+
+        return valid;
+    }
+
+    public Gtk.TreeRowReference? find_file_row (GOF.File file_a, GOF.Directory.Async? dir = null) {
+        var files = new GLib.Sequence<GOF.File> ();
+        files.append (file_a);
+        GLib.List<Gtk.TreeRowReference> result = find_file_rows (files, dir);
+        if (result == null) {
+            return null;
+        } else {
+            return result.data;
+        }
+    }
+
+    public GLib.List<Gtk.TreeRowReference> find_file_rows (GLib.Sequence<GOF.File> files, GOF.Directory.Async? dir = null) {
+        files.sort (file_match_func);  /* Sort in same order as model */
+
+        GLib.List<Gtk.TreeRowReference> rows_found = null;
+
+        GLib.SequenceIter<GOF.File> seq_iter = files.get_begin_iter ();
+        while (!seq_iter.is_end ()) {
+            GOF.File file_a = seq_iter.@get ();
+            @foreach ((model, path, iter) => {
+                GOF.File? file_b = null;
+                model.@get (iter, FM.ColumnID.FILE_COLUMN, out file_b);
+                if (file_b != null && !file_b.is_null) {
+                    if (file_a.location.equal (file_b.location)) {
+                        var row_ref = new Gtk.TreeRowReference (model, path);
+                        rows_found.prepend (row_ref);
+                        seq_iter = seq_iter.next ();
+                        if (seq_iter.is_end ()) {
+                            return true;
+                        } else {
+                            file_a = seq_iter.@get ();
+                        }
+                    }
+                }
+                return false;
+            });
+
+            seq_iter = seq_iter.next (); /* file was not in model */
+        }
+
+        return (owned)rows_found;
+    }
+
+    public new void get_value (Gtk.TreeIter iter, int column, out Value return_value) {
```

Comment:
Avoid the `new` keyword

Reply:
I am not sure it is necessary to override this method anyway as the model now only has one column.